addons/point_of_sale/tests/test_point_of_sale_flow.py

```diff
@@ -2313,3 +2313,83 @@ def test_change_is_deducted_from_cash(self):
+            'name': 'Partner 2',
+        })
+        test_product = self.env['product.product'].create({
+            'name': 'Test Product',
+            'available_in_pos': True,
+            'list_price': 1000,
+        })
+        self.pos_config.open_ui()
+        current_session = self.pos_config.current_session_id
+        current_session.update_stock_at_closing = True
+
+        # 2 orders, one with payment term, the other without
+        orders = self.PosOrder.create([
+            {
+                'company_id': self.env.company.id,
+                'session_id': current_session.id,
+                'partner_id': self.partner1.id,
+                'lines': [(0, 0, {
+                    'name': "OL/0001",
+                    'product_id': test_product.id,
+                    'price_unit': 1000.0,
+                    'qty': 1.0,
+                    'price_subtotal': 1000.0,
+                    'price_subtotal_incl': 1000,
+                })],
+                'amount_total': 1000.0,
+                'amount_tax': 0.0,
+                'amount_paid': 0.0,
+                'amount_return': 0.0,
+                'to_invoice': True,
+            }, {
+                'company_id': self.env.company.id,
+                'session_id': current_session.id,
+                'partner_id': partner2.id,
+                'lines': [(0, 0, {
+                    'name': "OL/0001",
+                    'product_id': test_product.id,
+                    'price_unit': 1000.0,
+                    'qty': 1.0,
+                    'price_subtotal': 1000.0,
+                    'price_subtotal_incl': 1000,
+                })],
+                'amount_total': 1000.0,
+                'amount_tax': 0.0,
+                'amount_paid': 0.0,
+                'amount_return': 0.0,
+                'to_invoice': True,
+            }
+        ])
+        for order in orders:
+            payment_context = {"active_ids": order.ids, "active_id": order.id}
+            order_payment = self.PosMakePayment.with_context(**payment_context).create({
+                'amount': order.amount_total,
+                'payment_method_id': self.cash_payment_method.id
+            })
+            order_payment.with_context(**payment_context).check()
+
+        order_with_term = self.env['pos.order'].search([('partner_id', '=', self.partner1.id)], limit=1)
+        invoice = self.env['account.move'].search([('invoice_origin', '=', order_with_term.name)], limit=1)
+        self.assertEqual(invoice.invoice_date + timedelta(days=10), invoice.invoice_date_due)
+
+        # If payment term is not set, the due date of the invoice is the date of the invoice
+        order_without_term = self.env['pos.order'].search([('partner_id', '=', partner2.id)], limit=1)
+        invoice = self.env['account.move'].search([('invoice_origin', '=', order_without_term.name)], limit=1)
+        self.assertEqual(invoice.invoice_date, invoice.invoice_date_due)
```

Comment:
```suggestion
        self.assertEqual(orders[0].account_move.invoice_date + timedelta(days=10), orders[0].account_move.invoice_date_due)

        # If payment term is not set, the due date of the invoice is the date of the invoice
        self.assertEqual(orders[1].account_move.invoice_date, orders[1].account_move.invoice_date_due)
```

With this new test you don't need to search your order/invoice, you have direct access to them.